src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -17,14 +17,19 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 	implements ToAPIApplicationCommandOptions
 {
 	public choices?: APIApplicationCommandOptionChoice[];
+	public autocomplete?: boolean;
```

Comment:
This should act like name and description and be typed as readonly (so you use the builder methods instead)

Reply:
This is fairly inconsistent across the package as stuff like `required` and `channelTypes` aren't readonly. Should I make them all readonly or just autocomplete?

Reply:
For this PR, do just autocomplete, and you can submit further PRs for those